### DISCUSSION TITLE
Remove the results getting using createquerybuilder

### DISCUSSION THREAD

**Discussion by shubhangi101092** [symfony/symfony#47923] [Q&A]
Hi ,

I have request Manager class which is inherting entitymanagerinterface.
 In my function I am getting results but not able to delete using remove function

return $this->requestRepository->createQueryBuilder('cat')
  | ->andWhere('cat.name = :searchTerm')
  | ->setParameter('searchTerm', $term)
  | ->getQuery()
  | ->getResults();
Now what to delete these records using remove function instead of delete.

How can I achieve that ?

**Comment by MatTheCat** [comment]
What do you mean? Why couldn't you use `EntityManagerInterface::remove`?

**Comment by ghost** [comment]
You have to call `flush` after calling `remove`.
